system/forms/widgets/jssnippetwidget.xml

```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<form i18nBaseUri="widgets.jssnippetwidget:">
+	<tab>
+		<fieldset>
+			<field sortorder="10" name="source_url"      required="true" />
```

Comment:
The Source_URL cannot be required if we optionally have internal_script. Also, would be good to separate these out into fieldsets with titles "External javascript" + "Inline javascript". All fields to go in the "External javascript" fieldset except for "internal_script" (which should be renamed to "inline_script").